src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -245,10 +245,20 @@ void clock_gpio_init_int_frac(uint gpio, uint src, uint32_t div_int, uint8_t div
         invalid_params_if(HARDWARE_CLOCKS, true);
     }
 
+#if !PICO_RP2040 // assert currently broken on RP2040, but we know that hardware has 16-bit integer part
+    static_assert(CLOCKS_CLK_GPOUT0_DIV_INT_MSB - CLOCKS_CLK_GPOUT0_DIV_INT_LSB == 15, "");
+#endif
+    invalid_params_if(HARDWARE_CLOCKS, div_int >> 16);
```

Comment:
I think that this `invalid_params_if` also needs to be inside the `#if !PICO_RP2040` block, because `div_int` is up to 24 bits on RP2040.